File: frontend/src/types/job.ts
import { User } from './user';

export type JobType = 'remote' | 'onsite' | 'hybrid';
export type JobStatus = 'open' | 'closed' | 'archived';

export interface Job {
  id: string;
  recruiter_id: string;
  title: string;
  description: string;
  salary?: number;
  location: string;
  type: JobType;
  status: JobStatus;
  created_at: string;
  updated_at: string;
  recruiter?: User;
}

export interface CreateJobRequest {
  title: string;
  description: string;
  salary?: number;
  location: string;
  type: JobType;
}

export interface UpdateJobRequest {
  title?: string;
  description?: string;
  salary?: number;
  location?: string;
  type?: JobType;
  status?: JobStatus;
}

export interface JobFilters {
  search?: string;
  location?: string;
  type?: JobType;
  salary_min?: number;
  salary_max?: number;
  status?: JobStatus;
  sort_by?: string;
  order?: 'asc' | 'desc';
  page?: number;
  limit?: number;
}

export interface JobListResponse {
  jobs: Job[];
  total: number;
  page: number;
  limit: number;
}

File: backend/internal/models/job.go
package models

import (
	"time"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type JobType string
type JobStatus string

const (
	JobTypeRemote JobType = "remote"
	JobTypeOnsite JobType = "onsite"
	JobTypeHybrid JobType = "hybrid"

	JobStatusOpen     JobStatus = "open"
	JobStatusClosed   JobStatus = "closed"
	JobStatusArchived JobStatus = "archived"
)

type Job struct {
	ID          uuid.UUID      `gorm:"type:uuid;primary_key;default:gen_random_uuid()" json:"id"`
	RecruiterID uuid.UUID      `gorm:"type:uuid;not null" json:"recruiter_id"`
	Title       string         `gorm:"not null" json:"title"`
	Description string         `gorm:"type:text;not null" json:"description"`
	Salary      *float64       `json:"salary,omitempty"`
	Location    string         `json:"location"`
	Type        JobType        `gorm:"type:varchar(20);not null" json:"type"`
	Status      JobStatus      `gorm:"type:varchar(20);default:'open'" json:"status"`
	CreatedAt   time.Time      `json:"created_at"`
	UpdatedAt   time.Time      `json:"updated_at"`
	DeletedAt   gorm.DeletedAt `gorm:"index" json:"-"`

	
	Recruiter    User          `gorm:"foreignKey:RecruiterID" json:"recruiter,omitempty"`
	Applications []Application `gorm:"foreignKey:JobID" json:"applications,omitempty"`
}

type JobResponse struct {
	ID          uuid.UUID    `json:"id"`
	RecruiterID uuid.UUID    `json:"recruiter_id"`
	Title       string       `json:"title"`
	Description string       `json:"description"`
	Salary      *float64     `json:"salary,omitempty"`
	Location    string       `json:"location"`
	Type        JobType      `json:"type"`
	Status      JobStatus    `json:"status"`
	CreatedAt   time.Time    `json:"created_at"`
	UpdatedAt   time.Time    `json:"updated_at"`
	Recruiter   *UserResponse `json:"recruiter,omitempty"`
}

func (j *Job) ToResponse(includeRecruiter bool) JobResponse {
	resp := JobResponse{
		ID:          j.ID,
		RecruiterID: j.RecruiterID,
		Title:       j.Title,
		Description: j.Description,
		Salary:      j.Salary,
		Location:    j.Location,
		Type:        j.Type,
		Status:      j.Status,
		CreatedAt:   j.CreatedAt,
		UpdatedAt:   j.UpdatedAt,
	}

	if includeRecruiter && j.Recruiter.ID != uuid.Nil {
		userResp := j.Recruiter.ToResponse()
		resp.Recruiter = &userResp
	}

	return resp
}
